WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -504,9 +504,20 @@ - (void)setRestorableSelectedIndexPath:(NSIndexPath *)restorableSelectedIndexPat
 {
     _restorableSelectedIndexPath = restorableSelectedIndexPath;
 
-    if (restorableSelectedIndexPath != nil) {
+    if (restorableSelectedIndexPath != nil && restorableSelectedIndexPath.section < [self.tableSections count]) {
         BlogDetailsSection *section = [self.tableSections objectAtIndex:restorableSelectedIndexPath.section];
-        self.selectedSectionCategory = section.category;
+        switch (section.category) {
```

Comment:
Is it possible for Quick Start or Domain Credit to be removed between launches? As a for-instance, if a user redeemed a domain credit in Calypso, or did Quick Start on another device? 

I don't understand the fine details of this code, but just wanted to check on that possibility.

Reply:
> Quick Start on another device

fwiw, this scenario isn't possible, as Quick Start doesn't sync between devices. The Quick Start menu should require the user to specifically dismiss it, however I'm not sure how that might interact with state restoration. Perhaps save state with QS, dismiss QS, then crash the app before it can save state?

Reply:
I played with Quick Start for a bit. I couldn't make it crash.

Reply:
I think that might be a scenario only with credit domain. Like save the state with a domain credit, make a purchase online and then open the app again. The stored index should be out of bounds. As @nheagy said not with QS.

Reply:
Wasn't so much thinking about a crash, as that it might restore in an invalid state if the index wasn't out of bounds, but wasn't valid either. If the index for domain credit is `2`, and we store that, then it goes away, and now "Site Pages" is at index `2`, we might be restoring the wrong thing?

Even if that _is_ the case, it's probably not critical right now though.

Reply:
>  If the index for domain credit is 2, and we store that, then it goes away, and now "Site Pages" is at index 2, we might be restoring the wrong thing?

With QS and Domain credit it shouldn't happen. But it can with another section, since the value stored is an index and not something else that can be addressed to the right section. We added a small map/inverse map to handle the dynamic section (Domain credit and QS) but it needs to be improved (and extended).